### ISSUE TITLE
NuGet 2.8.6 

### ISSUE THREAD

**Issue by mattroid** [fsprojects/IfSharp#77]
### Description

I'm trying to get my own nuget packages to work. They use the immutable libraries that require NuGet 2.8.6. 

I see the NuGet.dll in the bin for IFSharp with version 2.8.1 but I can't find a compatible Nuget.dll with version 2.8.6 anywhere to drop in. Where did you get the original? and can we update please?

**Comment by cgravill** [comment]
I wasn't the one who originally added this functionality. I took a look a while back and my working hypothesis is that it's actually NuGet.CommandLine

As a test I downloaded https://www.nuget.org/packages/NuGet.CommandLine/2.8.6 then directly referenced the NuGet.exe to replace NuGet.dll. This compiles, which is promising.

However, the kernel fails at runtime. The issue seems to be a 32bit/64bit mismatch. So needs investigating if it can be rebuilt (or adjusted) to match 64bit. Do you want to give it a go?

**Comment by mattroid** [comment]
I added NuGet.exe 2.8.6 to the references in the Kernal project and it builds fine. I can import Newtonsoft.Json, Yay it works! But... I'm trying to change the sources of the nuget repo and it doesn't appear to be using the ifsharp.exe.config? 

I add my repo url to the ifsharp.exe.config, the %home%/.nuget/NuGet.config, the %AppData%/NuGet/NuGet.config. Still can't find my custom packages. 

I tried to output the source on NugetManager.fs:161 but it doesn't show in the output of the running ipyhon.exe. 
I tried Console.Writeline
I tried source |> Display

Not sure how to print debug info from the Kernal project? Can you help

**Comment by cgravill** [comment]
I had a look at the code we use to install Nuget and we create a CustomInstallCommand:

```
let installer = CustomInstallCommand()
installer.Console <- NuGet.Common.Console()
installer.Arguments.Add(nugetPackage)
installer.OutputDirectory <- packagesDir
installer.Prerelease <- prerelease
```

Where that inherits from `NuGet.Commands.InstallCommand`

I think this corresponds to: https://docs.nuget.org/consume/command-line-reference#install-command

Where we could either set the Sources of the ConfigFile.

**Comment by angrydexterous** [comment]
@mattroid are you going to PR with the new ref? If not I'll do it

**Comment by cgravill** [comment]
It would be great to get an up to date version of the NuGet dependency.

We're also looking at adding support for specifying dependencies using Paket (https://github.com/fsprojects/Paket) but have run into some issues on clean machines.

**Comment by cgravill** [comment]
I tracked down and resolved the issues with Paket on clean machines - an extremely old version of Json.NET - and fixed it here: b63ab00339a44afffa4dc769561971014452dab6

@sylvanc has also made several improvements to Paket #86 #91 such that it's now a great way to get dependencies!

**Comment by sylvanc** [comment]
Excellent!
